java-squid/src/main/java/org/sonar/java/ast/parser/TreeFactory.java

```diff
@@ -1858,24 +1771,15 @@ public ExpressionTree newAnnotatedParameterizedIdentifier(
     return result;
   }
 
-  public NewArrayTreeImpl newArrayInitializer(InternalSyntaxToken openBraceToken, Optional<List<AstNode>> rests, InternalSyntaxToken closeBraceToken) {
+  public NewArrayTreeImpl newArrayInitializer(InternalSyntaxToken openBraceToken, Optional<List<Tuple<ExpressionTree, Optional<InternalSyntaxToken>>>> rests, InternalSyntaxToken closeBraceToken) {
     ImmutableList.Builder<ExpressionTree> initializers = ImmutableList.builder();
-    List<AstNode> children = Lists.newArrayList();
-
-    children.add(openBraceToken);
     if (rests.isPresent()) {
-      for (AstNode rest : rests.get()) {
-        initializers.add((ExpressionTree) rest.getFirstChild());
-        children.add(rest.getFirstChild());
-
-        if (rest.getNumberOfChildren() == 2) {
-          children.add(rest.getLastChild());
-        }
+      for (Tuple<ExpressionTree, Optional<InternalSyntaxToken>> rest : rests.get()) {
+        //FIXME SONARJAVA-547 commas should be handled.
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)